### PR TITLE
Feature/dhscft 1008 modular line chart over time

### PR DESCRIPTION
**Pull request by stuc-bjss**
# Description

Jira ticket: [DHSCFT-1008](https://bjss-enterprise.atlassian.net/browse/DHSCFT-1008)

Tackle tech debt by decoupling the Line Chart Over Time from the Views and prop drilling to it.

## Changes

- Use React Query to load data from the API for the chart
- Seed the query cache server side so that the UI doesn't make duplicate requests to the API 
- Propose new filing of charts into a charts dir

## Validation

Did not delete any existing tests but updated to make them work with new code
e2e tests pass without change

### CODE REVIEW

File: frontend/fingertips-frontend/components/charts/LineChartOverTime/helpers/lineChartOverTimeIsRequired.ts
Comment (stuc-bjss) [inline frontend/fingertips-frontend/components/charts/LineChartOverTime/helpers/lineChartOverTimeIsRequired.ts:1]: careful checking required here, I found the rules of when the line chart should be shown not very clear so hopefully this captures that correctly

File: frontend/fingertips-frontend/components/charts/LineChartOverTime/helpers/lineChartOverTimeRequestParams.ts
Comment (stuc-bjss) [inline frontend/fingertips-frontend/components/charts/LineChartOverTime/helpers/lineChartOverTimeRequestParams.ts:1]: what request params would the apiCient need to get data for the LineChartOverTime

File: frontend/fingertips-frontend/components/charts/LineChartOverTime/hooks/useLineChartOverTimeRequestParams.ts
Comment (stuc-bjss) [inline frontend/fingertips-frontend/components/charts/LineChartOverTime/hooks/useLineChartOverTimeRequestParams.ts:1]: as a hook get the api request params for a line chart over time from the search state hook

File: frontend/fingertips-frontend/components/hooks/useSearchStateParams.ts
Comment (stuc-bjss) [inline frontend/fingertips-frontend/components/hooks/useSearchStateParams.ts:1]: useMemo so that the return value is the SAME object reference if the searchParams have not changed

File: frontend/fingertips-frontend/components/atoms/SeedQueryCache/SeeQueryCache.test.tsx
Comment (williamwillis11) [inline frontend/fingertips-frontend/components/atoms/SeedQueryCache/SeeQueryCache.test.tsx:1]: think the filename has a type See instead of Seed

File: frontend/fingertips-frontend/components/atoms/SeedQueryCache/seedQueryCache.types.ts
Comment (williamwillis11) [inline frontend/fingertips-frontend/components/atoms/SeedQueryCache/seedQueryCache.types.ts:1]: seed lowercase in the filename in this one vs Seed capitalised in the others. was that on purpose?

File: frontend/fingertips-frontend/next.config.ts [hunk 1]
```diff
@@ -19,6 +19,14 @@ const nextConfig: NextConfig = {
   cacheHandler: require.resolve(
     'next/dist/server/lib/incremental-cache/file-system-cache.js'
   ),
+  async rewrites() {
+    return [
+      {
+        source: '/api/:path*', // your local path
+        destination: 'http://localhost:5144/:path*', // external API
```
Comment (WolfieKnee) [inline frontend/fingertips-frontend/next.config.ts:26]: Is it worth a code comment here to explain the purpose of this for dev?

File: frontend/fingertips-frontend/components/charts/LineChartOverTime/helpers/lineChartOverTimeData.test.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        healthData: mockHealthDataPoints([{ year: 2023 }, { year: 2024 }]),
+      }),
+    ],
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('returns chart data when one area has more than 1 time point', () => {
```
Comment (WolfieKnee) [inline frontend/fingertips-frontend/components/charts/LineChartOverTime/helpers/lineChartOverTimeData.test.ts:54]: add a test for 'returns null when there is only one time point'?

File: frontend/fingertips-frontend/lib/reactQueryClient.ts [hunk 3]
```diff
@@ -3,9 +3,9 @@ import { QueryClient } from '@tanstack/query-core';
 export const reactQueryClient = new QueryClient({
   defaultOptions: {
     queries: {
-      staleTime: Infinity,
       retry: false,
       refetchOnWindowFocus: false,
+      staleTime: 60 * 1000,
```
Comment (stuc-bjss) [inline frontend/fingertips-frontend/lib/reactQueryClient.ts:8]: magic numbers?

File: frontend/fingertips-frontend/components/charts/LineChartOverTime/helpers/lineChartOverTimeData.test.ts (hunk 2 above)
Comment (WolfieKnee) [reply frontend/fingertips-frontend/components/charts/LineChartOverTime/helpers/lineChartOverTimeData.test.ts:64]: covered in 'returns null when there are no valid time series data'

**Review by alibombaye — APPROVED** [review]
